sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2491,6 +2502,16 @@ class Dataset[T] private[sql](
     }
   }
 
+  /**
+   * Returns a new Dataset by updating an exisitng column with metadata.
+   *
+   * @group untypedrel
+   * @since 3.3.0
+   */
+  def withMetadata(existingColumnName: String, metadata: Metadata): DataFrame = {
+    withColumn(existingColumnName, col(existingColumnName), metadata)
```

Comment:
It would be nice to know the motivation here (maybe we can start with a JIRA ticket?). Adding metadata with `as` (`alias` in Python) is a bit verbose / repetitive but it is not that common operation anyway.